enhancements/console/customize-project-roles.md

```diff
@@ -0,0 +1,160 @@
+
+## Motivation
+
+Customers who uses their own kubernetes roles wants that the project admin can use this more specific options within the Developer Console.
+
+### Goals
+
+- Add configuration options to allow cluster admins to customize the project access roles.
+- Provide a list of default values to the admin
+
+### Non-Goals
+
+- Introducing a UI to configure this project access roles without a yaml editor.
+
+## Proposal
+
+### Extending the Console CRD
+
+Similar to [Customize Developer Catalog Categories](./catalog-categories.md) we extend the existing `operator.openshift.io/v1` / `Console` CRD. It provides already a `spec.customization` area where we can add a list of roles.
+
+The list of roles should link to `rbac.authorization.k8s.io/v1` / `Role` or `ClusterRole`.
+
+So the final yaml should look like this one:
+
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: Console
+metadata:
+  name: cluster
+  ...
+spec:
+  customization:
+    brand: online
+    projectAccess:
+    - admin
+    - edit
+    - view
+  ...
+```
+
+### Provide a code snippet in YAML editor
+
+The admin needs to know the current default roles to customize them. We don't want to provide the default roles in the CR so that we can change these defaults later without migrating the customer data.
+
+To provide the default values close to the editor, the console will provide a code snippet in the sidebar of the YAML editor when editing the `Console` operator config.
+
+### User Stories
+
+#### Story 1
+
+As an cluster admin, I want to customize (add, update or remove) some of the Developer Catalog project roles. The admin needs to know the current list of roles.
+
+#### Story 2
+
+As an project admin, I want use the customized roles to give other users access to the current project.
+
+### Risks and Mitigations
+
+#### How we handle upgrade
+
+**Mitigation**: We keep the default values as part of the console code, so that without customization the console always shows the default roles. If the default values are changed, the admin needs to merge the new values with their customization manually. The benefit of this is that we will never modify an already customized list.
+
+**Drawback** As soon as the admin customizes the roles, a cluster update will not add new default roles or drop none-existing roles.
+
+## Open Questions
+
+- [ ] Should we provide a way that roles could show a label (visible label in the select or dropdown for the user) instead of just the resource name.
+
+For example:
+
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: Console
+metadata:
+  name: cluster
+  ...
+spec:
+  customization:
+    brand: online
+    projectAccessRef:
+    - name: admin
+      label: Admin
+    - name: edit
+      label: Edit
+    - name: custom-role-name
+      label: "My custom role"
+  ...
+```
+
+- [ ] Should allow the admin to link `Role` or `ClusterRole`? Should this support other namespaces then the current?
```

Comment:
I think we should support both `Role` and `ClusterRole`.
@spadgett what exactly did 3.11 support? I want to say that it was a single list that performed a lookup by name in both cluster roles and roles lists.

The namespace will be implied based on the project in which the rolebinding will be created.

Reply:
@christianvogt check https://docs.openshift.com/container-platform/3.11/install_config/web_console_customization.html#changing-the-membership-whitelist which was done by [loading extension scripts](https://docs.openshift.com/container-platform/3.11/install_config/web_console_customization.html#changing-the-membership-whitelist).
Based on the docs, we only supported ClusterRole.